### PR TITLE
Restored / Added Terraform Linting

### PR DESCRIPTION
**Pull request by jaytmiller**
Restored github Terraform Validation actions and added subdir matrix
Updated github actions to Terraform version to 1.0.3.
Added tflint Terraform linting.
Tweaked ephemeral-setup/ecr.tf to address lint.

### CODE REVIEW

File: .github/workflows/terraform.yaml [hunk 1]
```diff
@@ -1,22 +1,59 @@
 name: Validate Terraform
+
 on:
+  - push
```
Comment (cslocum) [inline .github/workflows/terraform.yaml:4]: Curious why you enabled this?

File: .github/workflows/terraform.yaml [hunk 2]
```diff
@@ -1,22 +1,59 @@
...
+        subdir: [ephemeral-setup, one-time-setup, monitoring]
+
     steps:
-      - name: 'Checkout'
+      - name: Checkout
         uses: actions/checkout@master
-      - name: 'Terraform Init'
-        uses: hashicorp/terraform-github-actions@master
+
+      - name: Remove Backend Config
```
Comment (jmatuskey) [inline .github/workflows/terraform.yaml:22]: i guess if we decide to remove the s3 backends we should just codify that everywhere right?

**Review by cslocum — APPROVED** [review]
This is great!  Thanks.

File: .github/workflows/terraform.yaml [hunk 3]
```diff
@@ -1,22 +1,59 @@
...
+
 on:
+  - push
   - pull_request
+
 jobs:
   terraform:
     name: 'Terraform'
+
     runs-on: ubuntu-latest
```
Comment (cslocum) [inline .github/workflows/terraform.yaml:11]: Does this mean latest LTS?

**Review by jmatuskey — APPROVED** [review]

File: .github/workflows/terraform.yaml (hunk 1 above)
Comment (jaytmiller) [reply .github/workflows/terraform.yaml:4]: It's so the checks run whenever you push your current work up to a branch whether you PR or not.
It also causes the checks to run following a merge (which I think really has a "silent push" of the merged code).

File: .github/workflows/terraform.yaml (hunk 3 above)
Comment (jaytmiller) [reply .github/workflows/terraform.yaml:11]: No.  It's just a loosely specified Ubuntu environment for simple installs and shell commands

File: .github/workflows/terraform.yaml (hunk 2 above)
Comment (jaytmiller) [reply .github/workflows/terraform.yaml:22]: This is a cheat predicated on the idea that we're keeping S3 backends but they were causing issues for configuring the validation.    So the cheat just removes the backend config from our source code and validates as-if we use default backend handling with no explicit config.

File: .github/workflows/terraform.yaml (hunk 1 above)
Comment (cslocum) [reply .github/workflows/terraform.yaml:4]: I see, that makes sense.  Thanks.

**Pull request merged**